site/blog/a2a/components/A2AArchitectureVisualizer.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useState } from 'react';
+import { motion, AnimatePresence } from 'framer-motion';
+import styles from './A2AArchitectureVisualizer.module.css';
+
+interface AgentCard {
+  name: string;
+  capabilities: string[];
+  icon: string;
+}
+
+const agents: Record<string, AgentCard> = {
+  salesAgent: {
+    name: 'Sales Assistant',
+    capabilities: ['Lead qualification', 'Meeting scheduling', 'Quote generation'],
+    icon: '👩‍💼',
+  },
+  techAgent: {
+    name: 'Tech Support',
+    capabilities: ['Issue diagnosis', 'Code review', 'System monitoring'],
+    icon: '👨‍💻',
+  },
+};
+
+const communicationSteps = [
```

Comment:
Would like to rethink this animation, maybe we can draw it out as a flow chart or a graph instead of making people sit through each step.